### ISSUE TITLE
Need help with security issues

### ISSUE THREAD

**Issue by ownyo** [keystonejs/keystone#1425]
Hello,
i recently ran a security test on my webapllication and it pointed me to some risks. 
I am mainly missing some http headers:
- web browser xss protection not enabled
- x-content-type-options header missing
- x-frame-options header not set

Can somebody tell me how and where i can add those headers? Playing with the .htaccess didn't work.
Thanks in advance :)

**Comment by morenoh149** [comment]
http://expressjs.com/4x/api.html#res.append ?

**Comment by dcousens** [comment]
@ownyo perhaps use https://www.npmjs.com/package/helmet as an extra middleware?

**Comment by webteckie** [comment]
@dcousens not sure that's what you are implying but wouldn't keystonejs benefit from that helmet package to provide that security right out of the box?  It would have to expose it's configuration.

**Comment by VinayaSathyanarayana** [comment]
What are the plans to use helmet and the security best practices recommended for express applications at http://expressjs.com/en/advanced/best-practice-security.html 

Also please run http://cyh.herokuapp.com/cyh 
When I run the above on www.sydjs.com and demo.keystonejs.com, it shows a number of items to act upon.

Your views please: @JedWatson  @webteckie @jstockwin

**Comment by jstockwin** [comment]
This isn't an area I really know anything about, so will leave it to the other guys to comment.

**Comment by r3wt** [comment]
@jstockwin i know quite a bit about it. Personally i just place the node(s) behind nginx to solve this(as well as serving the static files for the node app as its much faster than node and frees node to do the important stuff), but i think it could make sense to optionally include helmet in keystone, and have it configurable via `set` or `init`. the question is on or off by default? i'm a fan of on by default as its probably the safest bet for less experienced developers. I have a feeling Jed will want to weigh in on this abit more. anyone who wants to read up about helmet can do so here: https://www.npmjs.com/package/helmet

**Comment by creynders** [comment]
I agree with @r3wt: on by default. People who'll have the need to turn it off will know how anyway.

Sidetrack: someone **cough _npm_ cough** needs to come up with a good solution for _really_ optional dependencies... I _really_ REALLY **really** wish an alternative would pop up, npm sucks big balls.

**Comment by VinayaSathyanarayana** [comment]
I ran OWASP ZAP on demo.keystonejs.com and got the following results. Please download the file rename it to html to see the results

[ksdemo1html.txt](https://github.com/keystonejs/keystone/files/311517/ksdemo1html.txt)

**Comment by VinayaSathyanarayana** [comment]
We should include `nsp check` as part of the build process.

**Comment by VinayaSathyanarayana** [comment]
Any update on including helmet?
